tests/test_full_workflow.py

```diff
@@ -443,3 +445,91 @@ async def test_failing_workflow_JobExecutionError(
         assert "JOB ERROR" in job_status_data["log"]
         assert "CANCELLED" in job_status_data["log"]
         assert "\\n" not in job_status_data["log"]
+
+
+async def test_non_python_task(
+    client,
+    MockCurrentUser,
+    testdata_path,
+    project_factory,
+    dataset_factory,
+    resource_factory,
+    tmp777_path,
+):
+    async with MockCurrentUser(persist=True) as user:
+        project_dir = tmp777_path / "test"
+        project = await project_factory(user, project_dir=str(project_dir))
+        payload = {"name": "WF", "project_id": project.id}
+        res = await client.post("api/v1/workflow/", json=payload)
+        workflow = res.json()
+        debug(workflow)
+        assert res.status_code == 201
+
+        file_in = f"{str(testdata_path)}/test_in.json"
+        file_out = f"{str(testdata_path)}/test_out.json"
+
+        if os.path.exists(file_in):
+            with open(file_in, "r") as f:
+                TEST_JSON = json.load(f)
+        else:
+            TEST_JSON = {"test": "json"}
+            with open(file_in, "w") as f:
+                json.dump(TEST_JSON, f)
+
+        task_dict = dict(
+            name="non-python",
+            source="custom task",
+            command=(
+                f"sh {str(testdata_path)}/issue189.sh "
+                f"--json {file_in} --metadata-out {file_out}"
```

Comment:
Remove `--json` and `--metadata-out` from `command`, which will be added by fractal-server.